inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -581,7 +581,14 @@ protected function get_sql_where_clause( $post_type ) {
 		global $wpdb;
 
 		$join   = '';
-		$status = "{$wpdb->posts}.post_status = 'publish'";
+		/**
+		 * Filter post status list for sitemap query for the post type.
+		 *
+		 * @param Array $post_status_names      Post status list, defaults to array( 'publish' ).
+		 * @param string $post_type Post type name.
+		 */
+		$post_status_names = apply_filters( 'wpseo_sitemap_poststatus' , array( 'publish' ), $post_type );
+		$status = "{$wpdb->posts}.post_status IN ('" . implode( "','", $post_status_names ) . "')";
```

Comment:
> Using `{$var}` is allowed from PHP 5.5. This needs to be changed to basic string concatenation.

According to [the PHP manual](http://php.net/manual/en/language.types.string.php#language.types.string.parsing.complex), "Functions, method calls, static class variables, and class constants inside {$} work since PHP 5." And the line I've changed already used {$wpdb->posts}.

Reply:
> The implode needs to be surrounded by a `esc_sql` to avoid sql injection.

I feel that anyone who could add code to filter the post status values could also run the sql themselves, but I have added the esc_sql as a safe guard.